### DISCUSSION TITLE
Not able to parse MDX: Expected a closing tag (..)before the end of `paragraph`

### DISCUSSION THREAD

**Discussion by vrepsys** [mdx-js/mdx#2241] [Q&A]
Hey fellas,

I'm having trouble parsing the following MDX content:
```
<details><summary>x</summary>
hello
</details>
``` 
(Playground link: https://mdxjs.com/playground/)

The error:  'Expected a closing tag for </summary> (1:1-1:10) before the end of paragraph  end-tag-mismatch  mdast-util-mdx-jsx'

When I change the input MDX to the following it starts working:
```
<details>
<summary>x</summary>
hello
</details>
```

However, I need to parse existing documents that are formatted in the way that throws an error. For example this MDX from Algolia's docs: https://raw.githubusercontent.com/algolia/docsearch/main/packages/website/docs/api.mdx

I thought perhaps I could prettify the MDX before passing it to the parser, but prettify doesn't always seems to format it correctly either: [Sandbox example](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJnGBDAlgGwGcA+VQgVwFtLsAnAT2PQjAGU46wALAFVuyiF82GLmiFUAegrU6jADpQMuAG7FFigAbaYhRZEEwABAA8kRgBpGAvEeCKjRgEbkYMaOftRHjs0YDkJv4ORgC+ADSKoQDcWtoaSpLoquqJmDgEJCDhIBAADqLiyKB0tBAA7gAKdAiEyCDY+OXY9HU5TvxgANZYrNiUcAAyuFBwyABmjYRw7Z09MKx52GAjAObIMLTkMyBwlE5w6JjogwKr5NircABiELQ0bmv12K4Q2SBcMJT4AOpcuPBCEswHB2IIAaoAfR6rhVlA7mMciNprQYJV+KsaBMpjsAFaEEysNb4OAARXIEHg2KIOyWtBR9Uo6BM7zytBGMB+uHQMC4yAAHAAGHJsiDTH78PL1NlwFEqREgACOFPg6PydRQ2EIAFpRodDu9aHBlbgjejLlikJMaTlppRcBstjtCMS4ABBNzslyquC0YajanTW2u8mUsZWnE5HBOLk8vlIABMUf4BDWAGEILJ6rKAKzvcjTHjYJwa61BkAqbYASSgmFgrDA7IKbtrrBg9BJgbgoVCQA)

Anyone has ideas how to approach this? Is there an option to relax the formatting rules or something like that?

Thank you

**Comment by wooorm** [comment]
I think you mean `prettier`, right? Otherwise I don’t know what you are talking about.

This behavior of what is and isn’t allowed is intentional. No way to change that.

Existing documents were likely written for MDX 1 (or just plain markdown). For MDX 2, they’ll have to be changed.
Prettier has currently open issues to add support for MDX 2

**Reply by vrepsys** [reply]
Got it. I guess I have to switch to MDX v1 or come up with some sort of solution to format JSX without prettier.

Btw, is there any documentation on how JSX should be formatted to parse properly in MDX v2?

**Reply by wooorm** [reply]
Yeah, on the website, [“What is MDX?”](https://mdxjs.com/docs/what-is-mdx/)

**Reply by wooorm** [reply]
You can format MDX files with `remark-cli` and `remark-mdx`, assuming both are installed: `remark **/*.mdx --use remark-mdx --output`

**Reply by vrepsys** [reply]
With this specific MDX content, I don't think I can format it with remark, because I assume formatting requires parsing. 

I tried formatting it with remark and it fails with the same error.

**Reply by wooorm** [reply]
You have things that look like MDX 1. Which fail currently.

remark expects MDX 2. I understood your question as asking about how to format MDX 2.

**Comment by joelhooks** [comment]
I had to migrate some documents from [mdx 1 to mdx 2 today and found the migration guide](https://mdxjs.com/migrating/v2/#update-mdx-files) to be helpful. Since this discussion is ranking for search, I thought I'd add a link https://mdxjs.com/migrating/v2/#update-mdx-files
